Source/core/CyclicBuffer.cpp

```diff
@@ -56,8 +56,34 @@ namespace Core {
             if (bufferSize != 0) {
 
                 #ifndef __WINDOWS__
-                _administration->_signal = PTHREAD_COND_INITIALIZER;
-                _administration->_mutex = PTHREAD_MUTEX_INITIALIZER;
+                pthread_condattr_t cond_attr;
+
+                // default values
+                int ret = pthread_condattr_init(&cond_attr);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                ret = pthread_condattr_setpshared(&cond_attr, PTHREAD_PROCESS_SHARED);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                ret = pthread_cond_init(&(_administration->_signal), &cond_attr);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                pthread_mutexattr_t mutex_attr;
+
+                // default values
+                ret = pthread_mutexattr_init(&mutex_attr);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                // enable access for threads, also in different processes
+                ret = pthread_mutexattr_setpshared(&mutex_attr, PTHREAD_PROCESS_SHARED);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+#ifndef __DEBUG__
```

Comment:
Shoudn't this be `#ifdef ___DEBUG___` ? (and as Pierre mentioned, see his arguments, not execute code in an ASSERT)

Reply:
Good catch. Indeed #ifdef and not #ifndef.

Reply:
In retrospect, if the remark is about the fact that the type influences the set of errors returned from locking and unlocking, and, this assert may hide it at those code points, that is correct. Fair point.